src/foundations/icon/Icon.props.ts

```diff
@@ -30,7 +32,8 @@ type PIcon = {
     onClick?: () => void;
 };
 
-export type PIconRoot = Required<Pick<PIcon, 'size' | 'color' | 'glyph'>> &
+export type PIconRoot = TransientProperty<Required<Pick<PIcon, 'size' | 'color'>>> &
```

Comment:
I am not sure i fully understand the concept. If a type abstraction is all that is necessary to omit the blockProperty can we just have more specifically defined types? Sorry if its a silly question.